src/lib/components/csvImportBox.svelte

```diff
@@ -27,37 +27,42 @@
     const importItems: Writable<ImportItemsMap> = writable(new Map());
 
     async function showCompletionNotification(
-        databaseId: string,
-        collectionId: string,
-        importData: Payload
+        database: string,
+        collection: string,
+        payload: Payload
     ) {
-        await invalidate(Dependencies.DOCUMENTS);
-        const url = `${base}/project-${page.params.region}-${page.params.project}/databases/database-${databaseId}/collection-${collectionId}`;
+        const isSuccess = payload.status === 'completed';
+        const isError = !isSuccess && !!payload.errors;
 
-        // extract clean message from nested backend error.
-        const match = importData.errors.join('').match(/message: '(.*)' Message:/i);
-        const errorMessage = match?.[1];
+        if (!isSuccess && !isError) return;
 
-        const type = importData.status === 'completed' ? 'success' : 'error';
-        const message =
-            importData.status === 'completed'
-                ? 'CSV import finished successfully.'
-                : `${errorMessage}`;
+        let errorMessage = 'Import failed. Check your CSV for correct fields and required values.';
+        if (isError && Array.isArray(payload.errors)) {
+            try {
+                // the `errors` is a list of json encoded string.
+                errorMessage = JSON.parse(payload.errors[0]).message;
```

Comment:
If we only do this for CSV import, how does regular migration UI look now?

Reply:
That should include the all the required details as those are fetched from the DB via an API call, not in realtime.

Reply:
Can you double check it displays okay with the errors as nested JSON